pkg/store/prometheus_test.go

```diff
@@ -420,7 +420,11 @@ func TestPrometheusStore_LabelValues_e2e(t *testing.T) {
 	u, err := url.Parse(fmt.Sprintf("http://%s", p.Addr()))
 	testutil.Ok(t, err)
 
-	proxy, err := NewPrometheusStore(nil, nil, promclient.NewDefaultClient(), u, component.Sidecar, getExternalLabels, nil)
+	version, err := promclient.NewDefaultClient().BuildVersion(ctx, u)
```

Comment:
Can you add a test case where the version is **malformed or incorrect**? I am not sure how would that behave, so that might be a good idea to add one and define the behaviour that we might want to do when version in **malformed**. 😄

Reply:
I am okay to have this in the next pr.